src/grt/src/GlobalRouter.cpp

```diff
@@ -924,6 +924,69 @@ void GlobalRouter::computeTrackConsumption(
   }
 }
 
+std::vector<int> GlobalRouter::findTransitionLayers()
+{
+  odb::dbTech* tech = db_->getTech();
+  int max_layer = std::max(max_routing_layer_, max_layer_for_clock_);
+  std::map<int, odb::dbTechVia*> default_vias = getDefaultVias(max_layer);
+  std::vector<int> transition_layers;
+  for (const auto [layer, via] : default_vias) {
+    odb::dbTechLayer* tech_layer = tech->findRoutingLayer(layer);
+    bool vertical = tech_layer->getDirection() == odb::dbTechLayerDir::VERTICAL;
+    int via_width = 0;
+    for (const auto box : default_vias[layer]->getBoxes()) {
+      if (box->getTechLayer()->getRoutingLevel() == layer) {
+        via_width = vertical ? box->getWidth() : box->getLength();
+        break;
+      }
+    }
+
+    int track_pitch = grid_->getTrackPitches()[layer - 1];
+    if ((static_cast<double>(via_width) / track_pitch) > 0.8) {
+      transition_layers.push_back(layer);
+    }
+  }
+
+  return transition_layers;
+}
+
+void GlobalRouter::adjustTransitionLayers(
+    const std::vector<int>& transition_layers,
+    std::map<int, std::vector<odb::Rect>>& layer_obs_map)
+{
+  odb::dbTech* tech = db_->getTech();
+  for (int layer : transition_layers) {
+    odb::dbTechLayer* tech_layer = tech->findRoutingLayer(layer);
+    for (const auto& obs : layer_obs_map[layer - 1]) {
+      odb::Rect first_tile_bds, last_tile_bds;
+      odb::Point first_tile, last_tile;
+      grid_->getBlockedTiles(
+          obs, first_tile_bds, last_tile_bds, first_tile, last_tile);
+      if (first_tile.x() != last_tile.x() || first_tile.y() != last_tile.y()) {
```

Comment:
I would suggest breaking out this condition into an early return.

```c++
if (first_tile.x() == last_tile.x() && first_tile.y() == last_tile.y()) {continue;}
```

That will help reduce the indentation here.